src/emitter.rs

```diff
@@ -260,6 +252,26 @@ impl<'a> YamlEmitter<'a> {
         }
     }
 
+    fn emit_literal_block(&mut self, v: &String) -> EmitResult {
+        let ends_with_newline = v.chars().last() == Some('\n');
+        if ends_with_newline {
+            write!(self.writer, "|")?;
+        } else {
+            write!(self.writer, "|-")?;
+        }
+
+        self.level += 1;
+        // lines() will omit the last line if it is empty.
+        for line in v.lines() {
+            writeln!(self.writer)?;
+            self.write_indent()?;
+            // It's literal text, so don't escape special chars.
+            write!(self.writer, "{line}")?;
```

Comment:
Can we call into `self.writer.write(line.as_bytes())` directly instead? That'll avoid going through the format machinery, which avoids the `"{line}"` stringification and associated memory allocations.

Ditto for the yaml-rust2 change ~ I think this might apply there as well.

Reply:
Fixed, but used `write_str()` instead.